server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/dto/VersionRequestTest.java

```diff
@@ -20,48 +20,41 @@
 package org.apache.james.webadmin.dto;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
-import org.junit.Rule;
+import org.apache.james.backends.cassandra.versions.SchemaVersion;
 import org.junit.Test;
-import org.junit.rules.ExpectedException;
 
 public class VersionRequestTest {
-    @Rule
-    public ExpectedException expectedException = ExpectedException.none();
-
     @Test
     public void parseShouldThrowWhenNullVersion() throws Exception {
-        expectedException.expect(NullPointerException.class);
-
-        CassandraVersionRequest.parse(null);
+        assertThatThrownBy(() -> CassandraVersionRequest.parse(null))
+            .isInstanceOf(NullPointerException.class);
     }
 
     @Test
     public void parseShouldThrowWhenNonIntegerVersion() throws Exception {
-        expectedException.expect(IllegalArgumentException.class);
-
-        CassandraVersionRequest.parse("NoInt");
+        assertThatThrownBy(() -> CassandraVersionRequest.parse("NoInt"))
+            .isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
     public void parseShouldThrowWhenNegativeVersion() throws Exception {
-        expectedException.expect(IllegalArgumentException.class);
-
-        CassandraVersionRequest.parse("-1");
+        assertThatThrownBy(() -> CassandraVersionRequest.parse("-1"))
+            .isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
     public void parseShouldAcceptZeroVersion() throws Exception {
-        CassandraVersionRequest cassandraVersionRequest = CassandraVersionRequest.parse("0");
-
-        assertThat(cassandraVersionRequest.getValue()).isEqualTo(0);
+        assertThatThrownBy(() -> CassandraVersionRequest.parse("0"))
```

Comment:
Why have you change this behavior?
(the method is wrong yet)

Reply:
Oh, well catched. I wanted to write a note about it (but I forgot).

The reason is consistency.

SchemaVersion = 0 is not usable (very clear with the SchemaVersion refactoring), the input should be rejected as close as possible from the UI.

BTW you see no test with version = 0 ;-)

Reply:
Ok, then just rename the test ;)